servant-client/src/Servant/Client/Internal/HttpClient.hs

```diff
@@ -153,15 +158,18 @@ instance Alt ClientM where
   a <!> b = a `catchError` \_ -> b
 
 instance RunClient ClientM where
-  runRequestAcceptStatus = performRequest
+  runRequestAcceptStatus statuses req = do
+    ClientEnv _ _ _ _ mid <- ask
+    let oldApp = performRequest statuses
+    mid oldApp req
   throwClientError = throwError
 
 runClientM :: ClientM a -> ClientEnv -> IO (Either ClientError a)
 runClientM cm env = runExceptT $ flip runReaderT env $ unClientM cm
 
 performRequest :: Maybe [Status] -> Request -> ClientM Response
 performRequest acceptStatus req = do
-  ClientEnv m burl cookieJar' createClientRequest <- ask
+  ClientEnv m burl cookieJar' createClientRequest _ <- ask
```

Comment:
I would like to suggest use of `namedFieldPuns` here too, but that's perhaps too big a change.

Reply:
I'd rather not change this one, I think it's not related enough.